core/lib/multivm/src/versions/vm_fast/vm.rs

```diff
@@ -86,6 +86,18 @@ impl VmRunResult {
     }
 }
 
+pub trait TracerExt: Tracer {
```

Comment:
Having to add this is not great. It could instead be added to the tracer interface, which is ok-ish. I don't want to duplicate the hook logic in the tracer. Do you have any other ideas?

Reply:
If the Vm knew where in the composite tracer ValidationTracer is, it could simply set its in_validation boolean.

I'm not a fan of the built-in CircuitsTracer, though. But I think I'll figure something out.